### PR TITLE
efi: Some minor cleanup

### PR DESCRIPTION
**Pull request by cgwalters**
- Don't need to serialize anymore
- Use `Default` since it's nicer

### CODE REVIEW

File: src/efi.rs [hunk 1]
```diff
@@ -24,15 +23,9 @@ use crate::util::CommandRunExt;
 /// The path to the ESP mount
 pub(crate) const MOUNT_PATH: &str = "boot/efi";
 
-#[derive(Serialize, Deserialize)]
+#[derive(Default)]
 pub(crate) struct EFI {}
 
-impl EFI {
-    pub(crate) fn new() -> Self {
```
Comment (lucab) [inline src/efi.rs:31]: What about repurposing this as a `pub(crate) fn new_component() -> Box<dyn Component>`?

File: src/bootupd.rs [hunk 2]
```diff
@@ -124,8 +124,9 @@ pub(crate) fn install(source_root: &str, dest_root: &str) -> Result<()> {
 pub(crate) fn get_components() -> Vec<Box<dyn Component>> {
     let mut components: Vec<Box<dyn Component>> = Vec::new();
 
+    // Unwrap safety - this component is built in for these targets
     #[cfg(any(target_arch = "x86_64", target_arch = "aarch64"))]
-    components.push(Box::new(efi::EFI::new()));
+    components.push(component::new_from_name("EFI").expect("efi"));
```
Comment (lucab) [inline src/bootupd.rs:129]: I did like the previous state of this. Why did you change the approach? Would the `EFI::new_component` suggestion below help?

Comment (cgwalters) [reply src/bootupd.rs]: Hm...is your objection the new unwrap?  I reworked it to call directly again.

File: src/efi.rs (hunk 1 above)
Comment (cgwalters) [reply src/efi.rs:31]: If I understand correctly, this doesn't seem much clearer to me than inlining the `Box::new(..:default())` - I reworked to do that.  Does that work?

File: src/bootupd.rs (hunk 2 above)
Comment (lucab) [reply src/bootupd.rs]: Not directly, I think it was overall on the "lookup via hardcoded string label". It's somehow subjective and minor, thus sorry I couldn't articulate it better. Current form looks good :+1:.

**Review by lucab — APPROVED** [review]
/lgtm
